src/sentry/static/sentry/app/components/idBadge/baseBadge.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import Avatar from 'app/components/avatar';
+import space from 'app/styles/space';
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+import {Team, Member, User, Organization, AvatarProject} from 'app/types';
+
+type Props = {
+  displayName: React.ReactNode;
+  user?: User;
+  member?: Member;
+  team?: Team;
+  organization?: Organization;
+  project?: AvatarProject;
+  description?: React.ReactNode;
+  avatarProps?: Record<string, any>;
+  avatarSize?: number;
+  hideAvatar?: boolean;
+  hideName?: boolean; // Hides the main display name
+  className?: string;
+};
+
+const BaseBadge = React.memo(
+  ({
+    hideName = false,
+    hideAvatar = false,
+    displayName,
+    description,
+    team,
+    organization,
+    project,
+    member,
+    user,
+    avatarProps = {},
+    avatarSize = 24,
+    className,
```

Comment:
Nit: It would be nice for these keys to have the same (or similar) order as the type for props.